src/app/components/splitter/splitter.ts

```diff
@@ -158,16 +158,13 @@ export class Splitter {
     }
 
     ngAfterContentInit() {
-        this.templates.forEach((item) => {
-            switch (item.getType()) {
-                case 'panel':
-                    this.panels.push(item.template);
-                    break;
-                default:
-                    this.panels.push(item.template);
-                    break;
-            }
+
+        this.templates.changes.subscribe(_ => {
```

Comment:
Hello, please add an unsubscribe.
Why do you use `detectChanges` and not `markForCheck`?